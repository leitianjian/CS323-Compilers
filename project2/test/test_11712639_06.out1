Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: test0
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: arg1
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: arg2
        RP
      CompSt (1)
        LC
        StmtList (2)
          Stmt (2)
            IF
            LP
            Exp (2)
              Exp (2)
                ID: arg1
              LT
              Exp (2)
                ID: arg2
            RP
            Stmt (2)
              CompSt (2)
                LC
                DefList (3)
                  Def (3)
                    Specifier (3)
                      TYPE: int
                    DecList (3)
                      Dec (3)
                        VarDec (3)
                          ID: a
                        ASSIGN
                        Exp (3)
                          INT: 10
                    SEMI
                StmtList (4)
                  Stmt (4)
                    RETURN
                    Exp (4)
                      ID: test1
                      LP
                      Args (4)
                        Exp (4)
                          ID: a
                      RP
                    SEMI
                RC
            ELSE
            Stmt (5)
              CompSt (5)
                LC
                StmtList (6)
                  Stmt (6)
                    RETURN
                    Exp (6)
                      INT: 2
                    SEMI
                RC
        RC
    ExtDefList (10)
      ExtDef (10)
        Specifier (10)
          TYPE: int
        FunDec (10)
          ID: test1
          LP
          VarList (10)
            ParamDec (10)
              Specifier (10)
                TYPE: int
              VarDec (10)
                ID: n
          RP
        CompSt (10)
          LC
          StmtList (11)
            Stmt (11)
              IF
              LP
              Exp (11)
                Exp (11)
                  ID: n
                LT
                Exp (11)
                  INT: 0
              RP
              Stmt (11)
                CompSt (11)
                  LC
                  StmtList (12)
                    Stmt (12)
                      RETURN
                      Exp (12)
                        ID: test0
                        LP
                        Args (12)
                          Exp (12)
                            INT: 1
                          COMMA
                          Args (12)
                            Exp (12)
                              INT: 3
                        RP
                      SEMI
                  RC
              ELSE
              Stmt (13)
                CompSt (13)
                  LC
                  DefList (14)
                    Def (14)
                      Specifier (14)
                        TYPE: char
                      DecList (14)
                        Dec (14)
                          VarDec (14)
                            ID: c
                          ASSIGN
                          Exp (14)
                            ID: test0
                            LP
                            Args (14)
                              Exp (14)
                                INT: 1
                              COMMA
                              Args (14)
                                Exp (14)
                                  INT: 2
                                COMMA
                                Args (14)
                                  Exp (14)
                                    INT: 3
                            RP
                      SEMI
                  StmtList (15)
                    Stmt (15)
                      Exp (15)
                        ID: test0
                        LP
                        Args (15)
                          Exp (15)
                            ID: c
                          COMMA
                          Args (15)
                            Exp (15)
                              INT: 2
                        RP
                      SEMI
                  RC
            StmtList (17)
              Stmt (17)
                RETURN
                Exp (17)
                  INT: 1
                SEMI
          RC
