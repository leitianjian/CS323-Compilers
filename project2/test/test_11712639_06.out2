Error type 2 at Line 4: test1 function is invoked without deﬁnition
Error type 8 at Line 4: function return type is not matched
Error type 9 at Line 14: the function’s arguments mismatch the declared parameters
Error type 5 at Line 14: unmatching types on both sides of assignment operator (=)
Error type 9 at Line 15: the function’s arguments mismatch the declared parameters
