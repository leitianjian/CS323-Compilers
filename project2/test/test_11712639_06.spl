int test0(int arg1, int arg2){
    if(arg1 < arg2){
        int a = 10;
        return test1(a);
    }else{
        return 2;
    }
}

int test1(int n){
    if(n < 0){
        return test0(1, 3);
    }else{
        char c = test0(1, 2, 3);
        test0(c, 2);
    }
    return 1;
}