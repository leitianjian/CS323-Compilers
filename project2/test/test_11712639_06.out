Error type 2 at line 4: test1 function is invoked without definition
Error type 9 at line 14: the function’s arguments mismatch the declared parameters
Error type 5 at line 14: unmatching types on both sides of assignment operator
Error type 9 at line 15: the function’s arguments mismatch the declared parameters
